### PR TITLE
[SerializerBundle] Support Yaml In Serializer Component

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/Symfony/Component/Serializer/Encoder/YamlDecode.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Encoder;
+
+use Symfony\Component\Yaml\Parser;
+
+class YamlDecode implements DecoderInterface
```
Comment (unkind) [inline src/Symfony/Component/Serializer/Encoder/YamlDecode.php]: I see no point to have both `YamlDecode`/`YamlEncode` and `YamlEncoder`. If user needs custom YAML encoder/decoder, he should create his `FooYamlEncoder implements EncoderInterface` and inject it.

The only case I see that user wants to replace all existing services with new implementation, but DIC has its own features for that, e.g. aliases.

File: src/Symfony/Component/Serializer/Encoder/YamlEncoder.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ */
+
+namespace Symfony\Component\Serializer\Encoder;
+
+class YamlEncoder implements EncoderInterface, DecoderInterface
+{
+    const FORMAT = 'yaml';
+
+    /**
+     * @var \QEResolv\Serializer\Encoder\Decode
```
Comment (unkind) [inline src/Symfony/Component/Serializer/Encoder/YamlEncoder.php]: What is this?

File: src/Symfony/Component/Serializer/Encoder/YamlEncoder.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    protected $decodeImpl;
+    
+    /**
+     * @var \QEResolv\Serializer\Encoder\Decode
```
Comment (unkind) [inline src/Symfony/Component/Serializer/Encoder/YamlEncoder.php]: Same here.

File: src/Symfony/Component/Serializer/Encoder/YamlEncode.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+use Symfony\Component\Yaml\Dumper;
+
+class YamlEncode implements EncoderInterface
+{
+
+    /**
+     * @var \Symfony\Component\Yaml\Dumper
+     */
+    protected $dumper;
```
Comment (stof) [inline src/Symfony/Component/Serializer/Encoder/YamlEncode.php]: internal properties should always be private in Symfony, unless we make an explicit choice for the opposite

File: src/Symfony/Component/Serializer/Encoder/YamlEncode.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        $this->dumper = $dumper ?: new Dumper();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function encode($data, $format, array $context = array())
+    {
+        return $this->dumper->dump((array) $data);
```
Comment (stof) [inline src/Symfony/Component/Serializer/Encoder/YamlEncode.php]: why casting to array ? this is wrong if the dumped data is a scalar

**Comment by dupuchba** [comment]
:+1:

File: src/Symfony/Component/Serializer/Encoder/YamlDecode.php (hunk 1 above)
Comment (dunglas) [inline src/Symfony/Component/Serializer/Encoder/YamlDecode.php]: I agree with @unkind

**Comment by dunglas** [comment]
Can you also make this new encoder available by default when enabling the Serializer in the full stack framework by patching FrameworkBundle?

**Comment by jakzal** [comment]
The original author of this PR has removed his account. If no one else is willing to continue his work we should close it.

**Comment by xabbuh** [comment]
@jakzal Seems he has a new account.

@georgerobinson Are you still interested in finishing this PR?

**Comment by fabpot** [comment]
Closing for now. @georgerobinson Feel free to reopen with your new account.

**Pull request closed without merging**